Tvl.VisualStudio.MouseFastScroll.IntegrationTests/Executor_InProc.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace Tvl.VisualStudio.MouseFastScroll.IntegrationTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using EnvDTE;
+
+    internal class Executor_InProc : InProcComponent
+    {
+        private Executor_InProc()
+        {
+        }
+
+        public static Executor_InProc Create()
+            => new Executor_InProc();
+
+        public object Execute(string assemblyFile, string typeName, string methodName, Dictionary<string, object> fieldValues)
+        {
+            var assembly = Assembly.LoadFrom(assemblyFile);
+            var type = assembly.GetType(typeName);
+            var obj = Activator.CreateInstance(type);
+            foreach (var field in type.GetFields())
+            {
+                var name = field.Name;
+                if (!name.StartsWith("<>"))
+                {
+                    var value = fieldValues[name];
+                    field.SetValue(obj, value);
+                }
+            }
+
+            var method = type.GetMethod(methodName, BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
+            var args = GetArguments(method);
+            return method.Invoke(obj, args);
```

Comment:
I'm wondering if for MBRO objects I should be doing the same as you in the `IntegrationService`?

```cs
// Create a unique URL for each object returned, so that we can communicate with each object individually
```

Could you explain what this does and the problem it solves? Does it also work when a COM object is returned?

Reply:
Given that the following test passes:
```cs
            [Fact]
            public void GetDte()
            {
                var dte = VisualStudio.ExecuteInHostProcess<DTE>(typeof(DteFactory), nameof(DteFactory.GetDte));
                Assert.NotNull(dte);
            }

            internal class DteFactory : InProcComponent
            {
                public static DTE GetDte()
                {
                    return GetDTE();
                }
            }
```

I'm guessing the answer is that I should be doing this!